src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,127 @@
+                USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                USER_MIN_ACCESSIBLE_AGE);
+        storageDao.add(user);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    private static Stream<Arguments> register_userData_ok() {
+        return Stream.of(
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_AGE_MORE_THAN_18_OK))
+        );
+    }
+
+    private static Stream<Arguments> register_userData_NotOk() {
+        return Stream.of(
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_LESS_6_SYMBOLS_NOT_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        USER_PASSWORD_LENGTH_LESS_6_SYMBOLS_NOT_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_AGE_LESS_18_NOT_OK)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_LESS_6_SYMBOLS_NOT_OK,
+                        USER_PASSWORD_LENGTH_LESS_6_SYMBOLS_NOT_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_LESS_6_SYMBOLS_NOT_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_AGE_LESS_18_NOT_OK)),
+                Arguments.of(new User(
+                        WHITE_SPACE,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        WHITE_SPACE,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        null,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        null,
+                        USER_MIN_ACCESSIBLE_AGE)),
+                Arguments.of(new User(
+                        USER_LOGIN_LENGTH_6_SYMBOLS_OK,
+                        USER_PASSWORD_LENGTH_6_SYMBOLS_OK,
+                        null)),
+                Arguments.of(new User(null, null, null))
+        );
+    }
+}
+
+
+
+
+
```

Comment:
don't forget to delete redundant lines at the end of the file.

Reply:
not resolved)